app/src/main/java/de/danoeh/antennapod/fragment/preferences/StoragePreferencesFragment.java

```diff
@@ -129,48 +144,72 @@ private void setupStorageScreen() {
-                                context.getResources().getText(R.string.send_label)));
-                    });
-                    alert.create().show();
-                }, error -> {
-                    alert.setTitle(R.string.export_error_label);
-                    alert.setMessage(error.getMessage());
-                    alert.show();
-                }, progressDialog::dismiss);
+                        buildExportSuccessDialog(context, context.getString(R.string.export_success_sum, output.toString()), fileUri);
+                    }, error -> {
+                        buildExportErrorDialog(context, error);
+                    }, progressDialog::dismiss);
+        } else {
+            Observable<DocumentFile> observable = new DocumentFileExportWorker(exportWriter, context, uri).exportObservable();
+            disposable = observable.subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(output -> {
+                        buildExportSuccessDialog(context, context.getString(R.string.export_success_sum, output.getUri()), output.getUri());
+                    }, error -> {
+                        buildExportErrorDialog(context, error);
+                    }, progressDialog::dismiss);
+        }
         return true;
     }
 
+    private void buildExportSuccessDialog(final Context context, final String message, final Uri streamUri) {
+        final AlertDialog.Builder alert = new AlertDialog.Builder(context)
+                .setNeutralButton(android.R.string.ok, (dialog, which) -> dialog.dismiss());
+        alert.setTitle(R.string.export_success_title);
+        alert.setMessage(message);
+        alert.setPositiveButton(R.string.send_label, (dialog, which) -> {
+            Intent sendIntent = new Intent(Intent.ACTION_SEND);
+            sendIntent.putExtra(Intent.EXTRA_SUBJECT,
+                    context.getResources().getText(R.string.opml_export_label));
+            sendIntent.putExtra(Intent.EXTRA_STREAM, streamUri);
+            sendIntent.setType("text/plain");
+            sendIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.KITKAT) {
+                List<ResolveInfo> resInfoList = context.getPackageManager().queryIntentActivities(sendIntent, PackageManager.MATCH_DEFAULT_ONLY);
+                for (ResolveInfo resolveInfo : resInfoList) {
+                    String packageName = resolveInfo.activityInfo.packageName;
+                    context.grantUriPermission(packageName, streamUri, Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                }
+            }
+            context.startActivity(Intent.createChooser(sendIntent,
+                    context.getResources().getText(R.string.send_label)));
+        });
+        alert.create().show();
+    }
+
+    private void buildExportErrorDialog(final Context context, final Throwable error) {
```

Comment:
Please rename to "showXyDialog". There also is the term of just building an alert dialog without showing it but that's not what happens here.

Reply:
Done!

Reply:
@ByteHamster can you have a look when you get a chance so we can get this merged? Thanks!